### PR TITLE
Kconfig advanced settings

### PR DESCRIPTION
**Pull request by boekholt**
I did not find a way to create a dependency between ENABLE_SDRAM and ENABLE_BRAM such that both options are always visible but only one of them is selectable.
Note: Additionally to the Kconfig changes I also modified ActionTypes.md since kconfig is taking care of consistency in the selection of options.

### CODE REVIEW

**Comment by boekholt** [comment]
We had this discussion yesterday between Jörg-Stephan, Thomas and myself -and I understood that our conclusion was to keep this option and to mark it as 'experimental'.
I would leave it that way because I see use cases (I am actually using it when building experimental NVMe models for simulation).

@jsvogt, @ThomasFuchs, @joergkayser: Any other opinions?

**Comment by jsvogt** [comment]
I agree. We still want to keep the BRAM option, and don't want to make configuration changes outside the config tool for such experimental options.

**Comment by bmesnet** [comment]
ok understand. If this stay as experimental/debug then i would suggest to push it away from the standard settings and put it in the advanced options - may be as the first item so it remains closed to the SDRAM to see it is deselected.

**Comment by boekholt** [comment]
I prefer to keep BRAM directly next to SDRAM because they are depending on each other. Furthermore, I'd like to have SDRAM next to the action because it is part of the action's configuration (instead of having the simulator choice between them).
Card type should be the first item followed by action because that is the main choice.
Therefore, I am not in favor of moving BRAM to the advanced section.

@jsvogt, @ThomasFuchs, @joergkayser: Your opinions regarding this?

File: scripts/Kconfig [hunk 1]
```diff
@@ -153,6 +141,18 @@ config SDRAM_USED
         default "TRUE"  if ENABLE_SDRAM
         default "FALSE" if ! ENABLE_SDRAM
 
+config ENABLE_BRAM
+        bool
+        prompt "Enable BRAM (replacing SDRAM for experimental use)"
+        depends on ! DISABLE_SDRAM_AND_BRAM
+        help
+          This is a debug option. The AXI attached on-card SDRAM will be replaced by 512KB BRAM.
```
Comment (jsvogt) [inline scripts/Kconfig]: debug -> experimental

**Pull request merged**